### PR TITLE
feat: implement time decay for vector search

### PR DESCRIPTION
**Pull request by noot**
## Description
previously, time decay (bias towards recent memories as opposed to older) was only implemented for lex search, not for vector search. this PR adds it to vector search.

## Type of Change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation update
- [ ] Performance improvement
- [ ] Code refactoring (no functional changes)

## Changes Made
  - added `recency_boost()` helper and `apply_recency_decay()` function in `src/memvid/search/helpers.rs` with shared `DEFAULT_DECAY_HALF_LIFE_SECS` constant (1 day)
  - applied exponential time decay reranking to `vec_search_with_embedding_acl()` in `src/memvid/search/api.rs`, using same formula as lex search
  - added configurable `decay_half_life_secs` parameter threaded through `vec_search_with_embedding_acl`, `search_adaptive`, and `search_adaptive_acl` (question: do we want this configurable, or just hard-coded to 1 day like in lex search?)
  - added `vec_decay_half_life_secs: Option<f32>` field to AskRequest so callers can tune the half-life per query (defaults to 1 day)
  - 10 unit tests covering decay curve properties, reranking behavior, edge cases

## Testing
- [x] I have added tests that prove my fix/feature works
- [x] All existing tests pass (`cargo test`)
- [x] I have tested on my local machine

## Documentation
- [x] I have updated relevant documentation
- [x] I have added doc comments for new public APIs
- [ ] CHANGELOG.md has been updated (if applicable)

## Checklist
- [x] My code follows the project's style guidelines
- [x] I have performed a self-review of my code
- [x] I have commented my code where necessary
- [x] My changes generate no new warnings
- [x] I have run `cargo clippy` and addressed any issues
- [x] I have run `cargo fmt` to format my code

### CODE REVIEW

File: src/types/ask.rs [hunk 1]
```diff
@@ -81,6 +81,10 @@ pub struct AskRequest {
     #[serde(default)]
     /// ACL evaluation mode (`audit` or `enforce`).
     pub acl_enforcement_mode: AclEnforcementMode,
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    /// Half-life in seconds for recency decay on vector search results.
+    /// Defaults to 86400 (1 day).
+    pub vec_decay_half_life_secs: Option<f32>,
```
Comment (ForeverAngry) [inline src/types/ask.rs]: Can we avoid making this a required field on the public `AskRequest` struct, or call this out as a breaking API change?

Serde defaults cover JSON/deserialization, but Rust callers using `AskRequest { ... }` literals will now fail to compile until they add `vec_decay_half_life_secs`. The behavior is good; the concern is just the construction contract for downstream users.

File: src/memvid/search/helpers.rs [hunk 2]
```diff
@@ -401,6 +401,48 @@ pub(crate) fn attach_temporal_metadata(memvid: &mut Memvid, hits: &mut [SearchHi
     Ok(())
 }
 
+pub(super) const DEFAULT_DECAY_HALF_LIFE_SECS: f32 = 86400.0;
+
+pub(super) fn recency_boost(age_seconds: f32, half_life_secs: f32) -> f32 {
+    let decay_factor = 2.0_f32.ln() / half_life_secs;
```
Comment (ForeverAngry) [inline src/memvid/search/helpers.rs:410]: Can we validate `decay_half_life_secs` before it reaches `recency_boost()`?

Values like `0.0`, negative numbers, `NaN`, or infinity can produce `NaN` scores or invert the decay behavior. A small guard that accepts only finite values greater than zero, plus a test for invalid inputs, would make this safer as a user-facing tuning knob.

**Review by ForeverAngry — CHANGES_REQUESTED** [review]
This direction makes sense to me. Vector search should observe the same recency behavior as lexical search, especially for memory-update workflows where newer facts need to win over stale context.

I’d like to see two small contract issues tightened before merge: the new public `AskRequest` field changes the Rust construction API, and the configurable half-life needs validation so invalid values can’t produce bad ranking scores.

File: src/types/ask.rs (hunk 1 above)
Comment (noot) [reply src/types/ask.rs]: removed this field, falls back to default decay time in the places it's used. let me know if that works or if you'd prefer to leave it and add a comment

File: src/memvid/search/helpers.rs (hunk 2 above)
Comment (noot) [reply src/memvid/search/helpers.rs:410]: added!

**Comment by noot** [comment]
@kehansama this PR adds the same logic that's already in lex search to vector search, I would say that a different time-decay algorithm is out of scope for this PR. however I agree that there are probably better algorithms than simple time decay, would be happy to look into them/implement them as an additional feature if there's buy-in from the maintainers
